### ISSUE TITLE
[Patch Request]: VFE Factions Expanded - Empire

### ISSUE THREAD

**Issue by ElvinDrude** [CombatExtended-Continued/CombatExtended#2409]
**Mod name: Vanilla Factions Expanded - Empire**

**Link to mod download (preferably Steam workshop or Github): https://steamcommunity.com/sharedfiles/filedetails/?id=2938820380**

**Description:**
This mod overhauls the Empire faction with new mechanics, quests, and equipment.

I'm unsure if this should be a patch request or a bug report - there seems to be an issue with CE and a single new piece of armour present in the mod: the [Janissary Helmet](https://github.com/Vanilla-Expanded/VanillaFactionsExpanded-Empire/blob/main/1.4/Defs/ThingDefs_Misc/Apparel_Headgear.xml#L211) reports the following XML error when you have both CE and this mod enabled:
```
XML error: Duplicate XML node name equippedStatOffsets in this XML block: 
<ThingDef ParentName="ArmorHelmetMakeableBase">
	<defName>VFEE_Apparel_JanissaryHelmet</defName>
	<label>janissary helmet</label>
	<description>A high-tech armored helmet with a built-in status computer and a protective plasteel-weave plate.\n\nIncludes a plasteel-plated gas mask that offers full protection against gas-related attacks, such as toxic environments.</description>
	<techLevel>Spacer</techLevel>
	<recipeMaker>
		<unfinishedThingDef>UnfinishedTechArmor</unfinishedThingDef>
		<researchPrerequisite>VFEE_JanissaryGear</researchPrerequisite>
		<skillRequirements>
			<Crafting>6</Crafting>
		</skillRequirements>
		<recipeUsers Inherit="False">
			<li>FabricationBench</li>
		</recipeUsers>
	</recipeMaker>
	<graphicData>
		<texPath>JanissaryArmor/JanissaryHelmet</texPath>
		<graphicClass>Graphic_Single</graphicClass>
		<drawSize>(0.9, 0.9)</drawSize>
	</graphicData>
	<tradeTags>
		<li>VFEE_Janissary</li>
	</tradeTags>
	<generateCommonality>2</generateCommonality>
	<statBases>
		<WorkToMake>16750</WorkToMake>
		<MaxHitPoints>190</MaxHitPoints>
		<MarketValue>645</MarketValue>
		<Mass>3.4</Mass>
		<Flammability>0.25</Flammability>
		<ArmorRating_Sharp>12</ArmorRating_Sharp>
		<ArmorRating_Blunt>26</ArmorRating_Blunt>
		<ArmorRating_Heat>0.46</ArmorRating_Heat>
		<Insulation_Cold>4</Insulation_Cold>
		<Insulation_Heat>2</Insulation_Heat>
		<EquipDelay>2</EquipDelay>
		<Bulk>4</Bulk>
		<WornBulk>1</WornBulk>
		<NightVisionEfficiency_Apparel>0.7</NightVisionEfficiency_Apparel>
	</statBases>
	<costList>
		<ComponentSpacer>1</ComponentSpacer>
		<Plasteel>50</Plasteel>
		<DevilstrandCloth>15</DevilstrandCloth>
	</costList>
	<apparel>
		<bodyPartGroups>
			<li>FullHead</li>
		</bodyPartGroups>
		<wornGraphicPath>JanissaryArmor/JanissaryHelmet</wornGraphicPath>
		<layers>
			<li>Overhead</li>
			<li>OnHead</li>
			<li>StrappedHead</li>
		</layers>
		<defaultOutfitTags>
			<li>Soldier</li>
		</defaultOutfitTags>
		<useDeflectMetalEffect>true</useDeflectMetalEffect>
		<tags>
			<li>VFEE_Janissary</li>
		</tags>
		<immuneToToxGasExposure>true</immuneToToxGasExposure>
	</apparel>
	<equippedStatOffsets>
		<ToxicEnvironmentResistance>1</ToxicEnvironmentResistance>
		<ToxicResistance>1</ToxicResistance>
	</equippedStatOffsets>
	<colorGenerator Class="ColorGenerator_Options">
		<options>
			<li>
				<weight>10</weight>
				<only>(0.9,0.9,0.9,1)</only>
			</li>
			<li>
				<weight>10</weight>
				<only>(0.33,0.33,0.33,1)</only>
			</li>
			<li>
				<weight>10</weight>
				<only>(0.65,0.65,0.65,1)</only>
			</li>
			<li>
				<weight>6</weight>
				<min>(0.3,0.3,0.3,1)</min>
				<max>(0.5,0.5,0.5,1)</max>
			</li>
		</options>
	</colorGenerator>
	<comps>
		<li Class="CompProperties_Biocodable" />
	</comps>
	<thingSetMakerTags>
		<li>RewardStandardHighFreq</li>
	</thingSetMakerTags>
	<equippedStatOffsets>
		<AimingAccuracy>0.15</AimingAccuracy>
		<ToxicEnvironmentResistance>0.50</ToxicEnvironmentResistance>
		<SmokeSensitivity>-1</SmokeSensitivity>
	</equippedStatOffsets>
	<modExtensions>
		<li Class="CombatExtended.PartialArmorExt">
			<stats>
				<li>
					<ArmorRating_Sharp>0.70</ArmorRating_Sharp>
					<parts>
						<li>Eye</li>
					</parts>
				</li>
				<li>
					<ArmorRating_Blunt>0.70</ArmorRating_Blunt>
					<parts>
						<li>Eye</li>
					</parts>
				</li>
				<li>
					<ArmorRating_Sharp>0.70</ArmorRating_Sharp>
					<parts>
						<li>Nose</li>
					</parts>
				</li>
				<li>
					<ArmorRating_Blunt>0.70</ArmorRating_Blunt>
					<parts>
						<li>Nose</li>
					</parts>
				</li>
				<li>
					<ArmorRating_Sharp>0.70</ArmorRating_Sharp>
					<parts>
						<li>Jaw</li>
					</parts>
				</li>
				<li>
					<ArmorRating_Blunt>0.70</ArmorRating_Blunt>
					<parts>
						<li>Jaw</li>
					</parts>
				</li>
			</stats>
		</li>
	</modExtensions>
</ThingDef>
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.XmlInheritance:CheckForDuplicateNodes (System.Xml.XmlNode,System.Xml.XmlNode)
Verse.XmlInheritance:ResolveXmlNodeFor (Verse.XmlInheritance/XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively (Verse.XmlInheritance/XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively (Verse.XmlInheritance/XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively (Verse.XmlInheritance/XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively (Verse.XmlInheritance/XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodesRecursively (Verse.XmlInheritance/XmlInheritanceNode)
Verse.XmlInheritance:ResolveXmlNodes ()
Verse.XmlInheritance:Resolve ()
Verse.LoadedModManager:ParseAndProcessXML (System.Xml.XmlDocument,System.Collections.Generic.Dictionary`2<System.Xml.XmlNode, Verse.LoadableXmlAsset>)
Verse.LoadedModManager:LoadAllActiveMods ()
Verse.PlayDataLoader:DoPlayLoad ()
Verse.PlayDataLoader:LoadAllPlayData (bool)
Verse.Root/<>c:<Start>b__6_1 ()
Verse.LongEventHandler:RunEventFromAnotherThread (System.Action)
Verse.LongEventHandler/<>c:<UpdateCurrentAsynchronousEvent>b__27_0 ()
System.Threading.ThreadHelper:ThreadStart_Context (object)
System.Threading.ExecutionContext:RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool)
System.Threading.ExecutionContext:Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool)
System.Threading.ExecutionContext:Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object)
System.Threading.ThreadHelper:ThreadStart ()

```

**Content:**
What does the new mod contain? Check all that apply.
- [X] New weapons.
- [ ] New animals, humanoids, or other creatures.
- [X] New factions.
- [X] New armor or apparel.

**Complete the following checklist before submitting a request.**
I hereby verify that I have done all of the following:
- [X] Searched the Steam workshop, Discord, and Github for an existing patch and not found one.
- [X] Confirmed there are no other open Patch Requests for this mod.
- [X] Confirmed the requested mod is not known to be fundamentally incompatible with Combat Extended.
- [X] **Have completed this template accurately and completely.**
- [X] **Understand that putting in a Patch Request does not guarantee a patch will be made.**
- [X] **Agree that I will not directly contact contributors to ask about patch requests.**

**Putting in request does not guarantee a patch will be made. Do not directly contact team members about Patch Requests. For smaller mods, consider reviewing the guide and making a patch yourself. If you wish to create a patch for any requested mod, there is no need to ask for permission.**

**Comment by ElvinDrude** [comment]
Looking more at this issue, I can see that this section of the patch is now incorrect: https://github.com/CombatExtended-Continued/CombatExtended/blob/Development/Patches/Vanilla%20Factions%20Expanded%20-%20Empire/Apparel_Headgear.xml#L297

As of this commit: https://github.com/Vanilla-Expanded/VanillaFactionsExpanded-Empire/commit/0891ffc379833be7eb32e3aff4e92fd219773839 there is now a defined `<equippedStatOffsets>` section so adding a second one is wrong. I unfortunately do not know what the right way to fix this is.

**Comment by SamaelGray** [comment]
There's already a pull request open to fix the issue, #2408.
